src/authHook.ts

```diff
@@ -228,3 +225,42 @@ export const getPluginConfigStartingCwd = (configuration: Configuration): Portab
   const isDlx = !!configuration.projectCwd?.endsWith(`dlx-${process.pid}`)
   return isDlx ? npath.toPortablePath(process.cwd()) : configuration.startingCwd
 }
+
+// Dependabot relies on this env variable so it's existence points to the fact
+// that we are running such an environment. See
+// https://github.com/dependabot/dependabot-core/blob/23de1c7583117bd15f26ab33482383ed57208f14/updater/lib/dependabot/environment.rb#L6
+//
+// Dependabot doesn't support passing environment variables. We can only pass the
+// token for the registry in a special dependabot.yaml config file. So this plugin
+// will never work for it, just skip the token calculation and configure it separately.
+const isRunningInDependabot = (): boolean => process.env.DEPENDABOT_JOB_ID !== undefined
+
+// In some environments you may want to use an existing auth token instead of fetching a new one
+// This is an escape hatch for that case
+const useExistingAuthtoken = (): boolean => process.env._YARN_PLUGIN_AWS_CODEARTIFACT_DISABLE !== undefined
+
+const shouldSkipPlugin = (): boolean => isRunningInDependabot() || useExistingAuthtoken()
```

Comment:
I tried to make these all vars, instead of functions; but tests sometimes had them set before the env vars were set; so I had to revert to functions.